src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -129,7 +129,19 @@ private bool EmitNupkg(PublishRoot root)
             if (root.Frameworks.Any())
             {
                 // Make sure we only emit the nupkgs for the specified frameworks
-                buildOptions.TargetFrameworks.AddRange(root.Frameworks);
+                // We need to pick actual project frameworks relevant to the publish
+                // but the project may not target exactly the right framework so we need
+                // to use a compatibility check.
+                foreach (var framework in root.Frameworks)
```

Comment:
File a todo to remove this logic because it isn't quite right. Results will end up varying between publish build and run. Basing publish on the lock file will resolve all of it.